src/output.ts

```diff
@@ -172,29 +188,41 @@ export default class Output {
 
 				for (let line of lines) {
 					const currentLine = output[y + offsetY];
-
 					// Line can be missing if `text` is taller than height of pre-initialized `this.output`
-					if (!currentLine) {
-						continue;
-					}
-
-					const width = stringWidth(line);
+					if (!currentLine) continue;
 
 					for (const transformer of transformers) {
 						line = transformer(line);
 					}
 
-					output[y + offsetY] =
-						sliceAnsi(currentLine, 0, x) +
-						line +
-						sliceAnsi(currentLine, x + width);
+					const chars = styledCharsFromTokens(tokenize(line));
+					let offsetX = x;
+					for (const char of chars) {
+						currentLine[offsetX] = char;
+
+						// Some characters take up more than one column. In that case, the following
+						// pixels need to be cleared to avoid printing extra characters
+						const charWidth = char.fullWidth ? 2 : char.value.length;
+						for (let dx = 1; dx < charWidth; dx++) {
+							currentLine[offsetX + dx] = {
+								type: 'char',
+								value: '',
+								fullWidth: false,
+								styles: char.styles
```

Comment:
maybe this should be
```suggestion
								styles: []
```
though.